uniffi/src/didcomm/from_prior.rs

```diff
@@ -0,0 +1,153 @@
+use std::sync::Arc;
+
+use didcomm_core::{error::ErrorKind, FromPrior as _FromPrior};
+
+use crate::DIDComm;
+
+use crate::common::{ErrorCode, EXECUTOR};
+use crate::did_resolver_adapter::DIDResolverAdapter;
+use crate::secrets_resolver_adapter::SecretsResolverAdapter;
+
+pub trait OnFromPriorPackResult: Sync + Send {
+    fn success(&self, from_prior_jwt: String, kid: String);
+    fn error(&self, err: ErrorKind, err_msg: String);
+}
+
+pub trait OnFromPriorUnpackResult: Sync + Send {
+    // TODO: return FromPriorExt
+    fn success(&self, from_prior: _FromPrior, kid: String);
+    fn error(&self, err: ErrorKind, err_msg: String);
+}
+
+pub struct FromPriorExt(Arc<_FromPrior>);
+
+impl FromPriorExt {
+    pub fn new(iss: String, sub: String, exp: Option<u64>) -> Self {
+        FromPriorExt(Arc::new(_FromPrior {
+            iss: iss,
+            sub: sub,
+            exp: exp,
+            aud: None,
+            iat: None,
+            jti: None,
+            nbf: None,
+        }))
+    }
+
+    pub fn get_iss(&self) -> String {
+        self.0.iss.clone()
+    }
+    pub fn get_sub(&self) -> String {
+        self.0.sub.clone()
+    }
+    pub fn get_exp(&self) -> Option<u64> {
+        self.0.exp
+    }
+}
+
+impl DIDComm {
+    pub fn pack_from_prior(
+        &self,
+        msg: &FromPriorExt,
+        issuer_kid: Option<String>,
+        cb: Box<dyn OnFromPriorPackResult>,
+    ) -> ErrorCode {
+        let msg = msg.0.clone();
+        let did_resolver = DIDResolverAdapter::new(self.did_resolver.clone());
```

Comment:
Do we need new adapter every time?

Reply:
Looks like yes (if we want to avid cloning DIDResolverAdapter), as didcomm core future expects a DIDResolver, not Arc<DIDResolver>.
We are cloning a reference (Arc) here, so it's not a problem.